src/main/java/com/myhome/controllers/AmenityController.java

```diff
@@ -87,4 +97,22 @@ public ResponseEntity deleteAmenity(@PathVariable String amenityId) {
         .map(ResponseEntity::ok)
         .orElse(ResponseEntity.notFound().build());
   }
+
+  @Operation(description = "Get all bookings for an amenity")
+  @GetMapping(path = "/amenities/{amenityId}/bookings")
+  public ResponseEntity<List<GetAmenityBookingsResponse>> getAmenitiesBookings(
+          @PathVariable String amenityId,
+          @RequestParam(required = false) LocalDateTime start,
+          @RequestParam(required = false) LocalDateTime end,
+          @PageableDefault(size = 200) Pageable pageable) {
+
+    List<AmenityBookingItem> amenityBookingItems =
+            amenitySDJpaService.listAllAmenityBookings(amenityId, start, end, pageable);
+    if (amenityBookingItems.isEmpty()) {
+      return ResponseEntity.notFound().build();
```

Comment:
404 not found should be returned only when `amenityId` not exists.
In case when there is `amenityId` but the query returns no results (there are no bookings for that amenity) we should return an empty array with 200 code

Reply:
done, just have one doubt. this way it will never return a 404 unless we do another query on the service when the list is empty to check if the amenityId exists. do you think it makes sense?

Reply:
Yes you are right, we will have then additional call, we have two ways:
- create HEAD endpoint which will do that operation (204 if resource exists, 404 if not exists) and we will call HEAD and GET requests on frontend side
- add one more database call inside GET endpoint which will cover that situation (if no records with given id, then return 404)

@jmprathab  What is your suggestion in this topic?

Reply:
done, thanks for the input